sweeper/aws_route53_resolver_firewall_rule: failing with `InvalidParameterException`

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Sweeper output
```
2021/10/13 14:11:28 [ERROR] Error running Sweeper (aws_route53_resolver_firewall_rule) in region (us-west-2): 1 error occurred:
	* error retrieving Route53 Resolver DNS Firewall rules: InvalidParameter: 1 validation error(s) found.
- missing required field, ListFirewallRulesInput.FirewallRuleGroupId.
```

--- a/aws/resource_aws_route53_resolver_firewall_rule_test.go
+++ b/aws/resource_aws_route53_resolver_firewall_rule_test.go
@@ -5,6 +5,7 @@ import (
 	"log"
 	"testing"
 
+	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/route53resolver"
 	"github.com/hashicorp/go-multierror"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
@@ -19,7 +20,7 @@ func init() {
 		Name: "aws_route53_resolver_firewall_rule",
 		F:    testSweepRoute53ResolverFirewallRules,
 		Dependencies: []string{
-			"aws_route53_resolver_firewall_rule_association",
+			"aws_route53_resolver_firewall_rule_group_association",
 		},
 	})
 }
@@ -32,35 +33,69 @@ func testSweepRoute53ResolverFirewallRules(region string) error {
 	conn := client.(*AWSClient).route53resolverconn
 	var sweeperErrs *multierror.Error
 
-	err = conn.ListFirewallRulesPages(&route53resolver.ListFirewallRulesInput{}, func(page *route53resolver.ListFirewallRulesOutput, lastPage bool) bool {
+	err = conn.ListFirewallRuleGroupsPages(&route53resolver.ListFirewallRuleGroupsInput{}, func(page *route53resolver.ListFirewallRuleGroupsOutput, lastPage bool) bool {
 		if page == nil {
 			return !lastPage
 		}
 
-		for _, firewallRule := range page.FirewallRules {
-			id := tfroute53resolver.FirewallRuleCreateID(*firewallRule.FirewallRuleGroupId, *firewallRule.FirewallDomainListId)
-
-			log.Printf("[INFO] Deleting Route53 Resolver DNS Firewall rule: %s", id)
-			r := resourceAwsRoute53ResolverFirewallRule()
-			d := r.Data(nil)
-			d.SetId(id)
-			err := r.Delete(d, client)
-
-			if err != nil {
-				log.Printf("[ERROR] %s", err)
-				sweeperErrs = multierror.Append(sweeperErrs, err)
+		for _, ruleGroup := range page.FirewallRuleGroups {
+			if ruleGroup == nil {
 				continue
 			}
+
+			ruleGroupId := aws.StringValue(ruleGroup.Id)
+
+			input := &route53resolver.ListFirewallRulesInput{
+				FirewallRuleGroupId: ruleGroup.Id,
+			}
+
+			err = conn.ListFirewallRulesPages(input, func(page *route53resolver.ListFirewallRulesOutput, lastPage bool) bool {
+				if page == nil {
+					return !lastPage
+				}
+
+				for _, firewallRule := range page.FirewallRules {
+					id := tfroute53resolver.FirewallRuleCreateID(*firewallRule.FirewallRuleGroupId, *firewallRule.FirewallDomainListId)
+
+					log.Printf("[INFO] Deleting Route53 Resolver DNS Firewall rule: %s", id)
+					r := resourceAwsRoute53ResolverFirewallRule()
+					d := r.Data(nil)
+					d.SetId(id)
+					err := r.Delete(d, client)
+
+					if err != nil {
+						log.Printf("[ERROR] %s", err)
+						sweeperErrs = multierror.Append(sweeperErrs, err)
+						continue
+					}
+				}
+
+				return !lastPage
+			})
+
+			if testSweepSkipSweepError(err) {
+				log.Printf("[WARN] Skipping Route53 Resolver DNS Firewall rules sweep (RuleGroup: %s) for %s: %s", ruleGroupId, region, err)
+				continue
+			}
+
+			if err != nil {
+				sweeperErrs = multierror.Append(sweeperErrs, fmt.Errorf("error retrieving Route53 Resolver DNS Firewall rules for rule group (%s): %w", ruleGroupId, err))
+				continue
+			}
+
+			return !lastPage
 		}
 
 		return !lastPage
 	})
+
 	if testSweepSkipSweepError(err) {
 		log.Printf("[WARN] Skipping Route53 Resolver DNS Firewall rules sweep for %s: %s", region, err)
 		return sweeperErrs.ErrorOrNil() // In case we have completed some pages, but had errors
 	}
+
 	if err != nil {
-		sweeperErrs = multierror.Append(sweeperErrs, fmt.Errorf("error retrieving Route53 Resolver DNS Firewall rules: %w", err))
+		sweeperErrs = multierror.Append(sweeperErrs, fmt.Errorf("error retrieving Route53 Resolver DNS Firewall rule groups: %w", err))
 	}
 
 	return sweeperErrs.ErrorOrNil()

